shotover/src/transforms/kafka/sink_cluster/connections.rs

```diff
@@ -227,23 +226,25 @@ impl Connections {
             .unwrap()
             .state
             .store(node_state, Ordering::Relaxed);
-        let connection = connection?;
-
-        // Recreating the node succeeded.
-        // So store it as the new connection, as long as we werent waiting on any responses in the old connection
-        let connection =
-            KafkaConnection::new(authorize_scram_over_mtls, sasl_mechanism, connection, None)?;
 
         if old_connection
             .map(|old| old.pending_requests_count())
             .unwrap_or(0)
             > 0
         {
-            Err(error.context("Succesfully reopened outgoing connection but previous outgoing connection had pending requests."))
-        } else {
-            self.connections.insert(destination, connection);
-            Ok(())
+            return Err(error.context("Outgoing connection had pending requests, those requests/responses are lost so connection recovery cannot be attempted."));
```

Comment:
this is moved before the `connection.context(..)?` to ensure that we return this error regardless of if the new connection is succesfully created or not.